knowledge-base/editor-mentions.md

```diff
@@ -0,0 +1,253 @@
+---
+title: Mentions in Editor
+description: Learn how to add support for mentions in the Telerik Editor component for Blazor.
+type: how-to
+page_title: Mentions in Editor
+slug: editor-kb-mentions
+position: 
+tags: telerik, blazor, editor, mentions
+ticketid: 1545505
+res_type: kb
+---
+
+## Environment
+
+<table>
+    <tbody>
+        <tr>
+            <td>Product</td>
+            <td>Editor for Blazor</td>
+        </tr>
+    </tbody>
+</table>
+
+## Description
+
+How to enable or implement support for @mentions in the TelerikEditor for Blazor, similar to GitHub, Facebook, etc.?
+
+## Solution
+
+You can use the [proseMirror-mentions](https://github.com/joelewis/prosemirror-mentions) plugin to provide a `@mentions` and `#hashtags` functionality for the Telerik Blazor Editor component. To implement the feature, customize and integrate the built-in [ProseMirror Schema](slug:editor-prosemirror-schema-overview) and [ProseMirror Plugins](slug:editor-prosemirror-plugins).
+
+Additionally, for proper dynamic positioning of the mentions list, you need to set the [`EditMode`](slug:Telerik.Blazor.Components.TelerikEditor#telerik_blazor_components_telerikeditor_editmode) property of the Editor to `EditorEditorMode.Div`. This ensures that the mentions dropdown can be positioned correctly relative to the editor content.
+
+````razor.skip-repl
+<TelerikEditor EditMode="EditorEditMode.Div">
+</TelerikEditor>
+````
+
+Setup steps:
+1. Create a directory at the root of your project which will contain your javascript module (e.g. `NpmJs`)
+2. In that directory, run
+> `npm init`
+3. Install a javascript bundler, in this example we'll be using [webpack](https://webpack.js.org), so run 
+> `npm install webpack webpack-cli --save-dev`
+4. Add an `npm build` script that will use webpack to bundle our Javascript file. Modify the `scripts` section of the `package.json` file to add the following build script (note: in this example all of our Javascript files will go into `NpmJs/src`)
+````json.skip-repl
+"scripts": {
+    "build": "webpack ./src/index.js --output-path ../wwwroot/js --output-filename index.bundle.js"
+  },
+````
+5. Install [proseMirror-mentions](https://github.com/joelewis/prosemirror-mentions) by running
+> `npm install prosemirror-mentions`
+
+The following code demonstrates how to integrate the `proseMirror-mentions` plugin in the Editor.
+
+<div class="skip-repl"></div>
+
+````razor Component
+@using Microsoft.Extensions.Logging.Abstractions
+
+@implements IDisposable
+
+<TelerikEditor Plugins="pluginsProvider"
+               Schema="schemaProvider"
+               EditMode="EditorEditMode.Div">
+</TelerikEditor>
+
+@code {
+    private DotNetObjectReference<Editor>? _dotNetRef;
```

Comment:
Underscores generate compiler warnings and may be out of fashion now. Clarify what Editor is and what to do with it.

```suggestion
    // Replace Home with your actual component type
    private DotNetObjectReference<Home>? dotNetRef;

```